common/config.go

```diff
@@ -32,76 +30,86 @@ type MoneroNode struct {
 
 // Config contains constants that are defaults for various environments
 type Config struct {
+	Env                      Environment
 	DataDir                  string
 	MoneroNodes              []*MoneroNode
-	ContractAddress          ethcommon.Address
+	SwapFactoryAddress       ethcommon.Address
 	ForwarderContractAddress ethcommon.Address
 	Bootnodes                []string
 }
 
 // MainnetConfig is the mainnet ethereum and monero configuration
-var MainnetConfig = Config{
-	DataDir: path.Join(baseDir, "mainnet"),
-	MoneroNodes: []*MoneroNode{
-		{
-			Host: "node.sethforprivacy.com",
-			Port: 18089,
+func MainnetConfig() *Config {
+	return &Config{
```

Comment:
There are a bunch of places in the code where I removed global and package level variables that we modify, so multiple swapd instances (simultaneous or successive) do not modify each other's state.